android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -39,9 +42,20 @@ object WorkerBinder {
    * @param worker The class that wants to be informed when to start and stop doing work.
    * @return [WorkerUnbinder] to unbind [Worker&#39;s][Worker] lifecycle.
    */
+  @OptIn(DelicateCoroutinesApi::class)
   @JvmStatic
-  open fun bind(interactor: Interactor<*, *>, worker: Worker): WorkerUnbinder {
-    return bind(mapInteractorLifecycleToWorker(interactor.lifecycle()), worker)
+  public fun bind(interactor: Interactor<*, *>, worker: Worker): WorkerUnbinder {
+    val workerEvents = interactor.lifecycleFlow.map { it.asWorkerEvent() }
+    // We need to launch in Unconfined dispatcher, AND start undispatched.
+    // If we only use Unconfined, and `onStart` launches some coroutine on Unconfined, coroutine will be put on an event loop.
+    val job = GlobalScope.launch(RibDispatchers.Unconfined, start = CoroutineStart.UNDISPATCHED) {
+      workerEvents
+        .takeWhile { it == WorkerEvent.START }
+        .onEach { worker.onStart(WorkerScopeProvider(workerEvents)) }
```

Comment:
Let add `distinctUntilChanged` between just to make sure that onStart will not be called multiple times somewhy.

Reply:
I don't think it can happen. Actually, this should never happen, so maybe we better raise an exception if it ever happens, instead of just filtering it out, as it means programmer error. Thanks for this suggestion, I'll reflect on that a bit more. But notice current implementation does not filter out values whatsoever.

Reply:
👍